server/server.go

```diff
@@ -0,0 +1,60 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockery --name=StorageAdapter --inpackage --filename=storage_adapter_mock.go
+type StorageAdapter interface {
+	StoreHostTelemetry(*HostTelemetry) error
+}
+
+type HostTelemetry struct {
+	AgentID       string    `json:"agent_id"`
+	SLESVersion   string    `json:"sles_version"`
+	CPUCount      int       `json:"payload"`
+	SocketCount   int       `json:"socket_count"`
+	TotalMemoryMB int       `json:"total_memory_mb"`
+	CloudProvider string    `json:"cloud_provider"`
+	Time          time.Time `json:"time"`
+}
+
+func hostTelemetryHandler(adapters ...StorageAdapter) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		var hostTelemetry HostTelemetry
+		err := json.NewDecoder(r.Body).Decode(&hostTelemetry)
+		if err != nil {
+			log.Errorf("Error unmarshaling host telemetry: %v", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		log.Debugf("Received new host telemetry data from agent: %s", hostTelemetry.AgentID)
+		for _, adapter := range adapters {
```

Comment:
Why do we need multiple `adapters` here?
I don't see the use case right now as we are only storing host telemetry

Reply:
@arbulu89 we are going to add the postgresql adapter in the following PRs, we still store the host telemetry but in another database